### ISSUE TITLE
Deb artifacts should be GPG signed

### ISSUE THREAD

**Issue by bsiegel** [sbt/sbt-launcher-package#72]
Deb artifacts should be GPG signed so they can be installed without warnings or workarounds on Debian/Ubuntu systems. [sbt-native-packager supports the `debianSign` action](https://github.com/sbt/sbt-native-packager/blob/master/src/main/scala/com/typesafe/sbt/packager/debian/NativePackaging.scala#L57), and [bintray supports GPG signing via their REST API](https://bintray.com/docs/api.html#_signing).

**Comment by jsuereth** [comment]
Yeah, signing would require somehow installing our PGP key into your apt-get keys as well though.   It's something we want to do, just need to find time to get all the install instructions accurate after the move.

**Comment by eed3si9n** [comment]
Related sbt/website#132?

**Comment by benmccann** [comment]
Yes, please! We'll just need one extra line in the install instructions. We can probably store the key on the Ubuntu keyserver. E.g. here's what's in the [MongoDB installation docs](http://docs.mongodb.org/manual/tutorial/install-mongodb-on-ubuntu/):

    sudo apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 --recv 7F0CEB10

**Comment by benmccann** [comment]
Btw, users get warnings today about installing sbt because the packages aren't signed:

>WARNING: The following packages cannot be authenticated!
  sbt
Install these packages without verification? [y/N]

**Comment by Blaisorblade** [comment]
> Btw, users get warnings today about installing sbt because the packages aren't signed:
> 
> WARNING: The following packages cannot be authenticated!
> sbt
> Install these packages without verification? [y/N]

Let me rephrase that to "following install instructions is a security risk". I decided to rather install the DEB manually from the HTTPS link.

> We can probably store the key on the Ubuntu keyserver. E.g. here's what's in the MongoDB installation docs:
> 
> sudo apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 --recv 7F0CEB10

PPAs (Personal Package Archives) save even that step, but maybe they require extra setup (though I guess not overly so).

But anyway, I'd mainly just like to vote for this issue, not to complain.

**Comment by eed3si9n** [comment]
Since I haven't announced 0.13.9 publicly, I republished 0.13.9 deb with autosigning on Bintray.
Here's how to import sbt's public key on your machine:

```
$ sudo apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 --recv 642AC823
...
gpg: requesting key 642AC823 from hkp server keyserver.ubuntu.com
gpg: key 642AC823: public key "sbt build tool <scalasbt@gmail.com>" imported
gpg: Total number processed: 1
gpg:               imported: 1  (RSA: 1)

$ sudo apt-get install sbt
....
The following NEW packages will be installed:
  sbt
0 upgraded, 1 newly installed, 0 to remove and 168 not upgraded.
Need to get 1,034 kB of archives.
After this operation, 1,226 kB of additional disk space will be used.
Get:1 https://dl.bintray.com/sbt/debian/  sbt 0.13.9 [1,034 kB]
Fetched 1,034 kB in 7s (144 kB/s)                                              
Selecting previously unselected package sbt.
(Reading database ... 730485 files and directories currently installed.)
Preparing to unpack .../archives/sbt_0.13.9_all.deb ...
Unpacking sbt (0.13.9) ...
Processing triggers for man-db (2.6.7.1-1ubuntu1) ...
Setting up sbt (0.13.9) ...
```

Try it out and let me know if it works.

**Comment by benmccann** [comment]
Seems like it worked!

**Comment by eed3si9n** [comment]
@benmccann Thanks for the patience, and poking us to do the right thing :)

**Comment by czchen** [comment]
@eed3si9n 

Could you use a more official email id for PGP key (ex: build@scala-sbt.org or the like) to sign package? `scalasbt@gmail.com` does not look like an official email from scala-sbt.

**Comment by younes-abouelnagah** [comment]
I second @czchen. I had to search for the key on Google to make sure that it is really related to SBT and that the page was not altered to install malware.

**Comment by ePaul** [comment]
Also, the key 642AC823 doesn't seem to have any signature by anyone else, i.e. it is not linked in the web of trust. How should people trust this key?

**Comment by quentinous** [comment]
Doesn't work at all....
